registry/docker/client.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package docker
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/tsuru/config"
+)
+
+func (r *dockerRegistry) doRequest(method, path string, headers map[string]string) (*http.Response, error) {
+	endpoint := fmt.Sprintf("http://%s%s", r.server, path)
+	req, err := http.NewRequest(method, endpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+	resp, err := r.HTTPClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func (r *dockerRegistry) client() (err error) {
```

Comment:
maybe use one of the clients in https://github.com/tsuru/tsuru/blob/master/net/client.go#L37?